libcxx/include/__expected/expected.h

```diff
@@ -893,11 +854,15 @@ class expected {
   }
 
 private:
-  struct __empty_t {};
```

Comment:
@huixie90 Do you remember what was the purpose of this `__empty` member in the union?

Reply:
I think the intent is that when the union is value-initialized, we can only zero-initialize the `__empty_` member whose size is only 1 byte,

Reply:
And previously, this was needed when the `expected` was initialized like this (pseudo code):

```c++
expected(const expected& other) : union_(), has_value_(other.has_value_) {
  // `std::construct_at` into `union_`
}
```

...but now, the union is never value-initialized any longer. All `expected` constructors will either initialize value or error, and make use of the member initializer list:

```c++
expected(const expected& other) : union_(other.has_value_, other.union_), has_value_(other.has_value_) { }
```

Reply:
Thanks!